src/backend/renderer/mod.rs

```diff
@@ -168,26 +168,62 @@ pub trait Frame {
         &mut self,
         texture: &Self::TextureId,
         matrix: Matrix3<f32>,
+        verticies: [Vector2<f32>; 4],
         alpha: f32,
     ) -> Result<(), Self::Error>;
+
     /// Render a texture to the current target as a flat 2d-plane at a given
-    /// position, applying the given transformation with the given alpha value.
+    /// position, cropping at the given offsets and applying the given transformation with the given alpha value.
     ///
     /// This operation is only valid in between a `begin` and `finish`-call.
     /// If called outside this operation may error-out, do nothing or modify future rendering results in any way.
     fn render_texture_at(
         &mut self,
         texture: &Self::TextureId,
         pos: (i32, i32),
+        scale: u32,
+        transform: Transform,
+        alpha: f32,
+    ) -> Result<(), Self::Error> {
+        let (w, h) = texture.size();
+        self.render_texture_from_to(
+            texture,
+            Rectangle {
+                x: 0,
+                y: 0,
+                width: w as i32,
+                height: h as i32,
+            },
+            Rectangle {
+                x: pos.0,
+                y: pos.1,
+                width: (w * scale) as i32,
+                height: (h * scale) as i32,
+            },
+            transform,
+            alpha,
+        )
+    }
+
+    /// Render part of a texture as given by src to the current target into the rectangle described by dest
+    /// as a flat 2d-plane after applying the given transformations.
+    ///
+    /// This operation is only valid in between a `begin` and `finish`-call.
```

Comment:
Are the comments about `begin`/`finish` still relevant with the render closure-based API?